CRM/Civicase/Service/CaseTypeCategoryProcessor.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+use CRM_Civicase_Service_CaseCategoryMenu as CaseCategoryMenuService;
+use CRM_Civicase_Service_CaseCategoryCustomDataType as CaseCategoryCustomDataType;
+use CRM_Civicase_Service_CaseCategoryCustomFieldExtends as CaseCategoryCustomFieldExtends;
+
+/**
+ * Class CRM_Civicase_Service_CaseTypeCategoryProcessor.
+ */
+class CRM_Civicase_Service_CaseTypeCategoryProcessor {
```

Comment:
The following public methods in this class, createMenuItem, updateMenuItem, deleteMenuItem does not reflect their names. For example the createMenuItem function creates a menu , adds custom data and add custom field extends option value. This does not follow the Single responsibility principle.